packages/field-base/src/field-mixin.js

```diff
@@ -97,55 +93,94 @@ export const FieldMixin = (superclass) =>
+          }
+        });
+      });
     }
 
     /** @protected */
     ready() {
       super.ready();
 
-      if (this._errorNode) {
-        this._errorNode.id = this._errorId;
+      const error = this._errorNode;
+      if (error) {
+        error.id = this._errorId;
 
-        this._applyCustomError();
+        this.__applyCustomError();
 
         this._updateErrorMessage(this.invalid, this.errorMessage);
       }
 
-      if (this._helperNode) {
-        this._currentHelper = this._helperNode;
-        this._helperNode.id = this._helperId;
-
-        this._applyCustomHelper();
-
-        this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
-        this.__helperSlot.addEventListener('slotchange', this.__onHelperSlotChange.bind(this));
-        this.__helperIdObserver = new MutationObserver((mutationRecord) => {
-          mutationRecord.forEach((mutation) => {
-            // only handle helper nodes
-            if (
-              mutation.type === 'attributes' &&
-              mutation.attributeName === 'id' &&
-              mutation.target === this._currentHelper &&
-              mutation.target.id !== this.__savedHelperId
-            ) {
-              this.__updateHelperId(mutation.target);
-            }
-          });
-        });
-
-        this.__helperIdObserver.observe(this, { attributes: true, subtree: true });
+      const helper = this._helperNode;
+      if (helper) {
+        this.__applyCustomHelper(helper);
       }
+
+      this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
+
+      this._observer = new FlattenedNodesObserver(this.__helperSlot, (info) => {
+        const helper = this._currentHelper;
+
+        const newHelper = info.addedNodes.find((node) => node !== helper);
+        const oldHelper = info.removedNodes.find((node) => node === helper);
+
+        if (newHelper) {
+          // Custom helper is added, remove the default one.
+          if (helper && helper.isConnected) {
+            this.removeChild(helper);
+          }
+
+          this.__applyCustomHelper(newHelper);
+
+          this.__helperIdObserver.observe(newHelper, { attributes: true, subtree: true });
+        } else if (oldHelper && this.helperText) {
+          // Custom helper is removed, restore the default one.
+          const helper = this.__defaultHelper;
+          helper.textContent = this.helperText;
+          this.__applyDefaultHelper(helper);
+        }
+      });
     }
 
-    /** @protected */
-    _applyCustomError() {
+    /** @private */
+    __applyCustomError() {
       const error = this.__errorMessage;
       if (error && error !== this.errorMessage) {
         this.errorMessage = error;
         delete this.__errorMessage;
       }
     }
 
+    /** @private */
+    __applyCustomHelper(helper) {
+      this.__updateHelperId(helper);
+
+      const helperText = helper.textContent;
+      if (helperText !== this.helperText) {
+        this.helperText = helperText;
```

Comment:
This feels like an unexpected side-effect. If you use a helper component, the derived `textContent` may be something unexpected, including line-break characters etc. Is there a reason a slotted helper should update the property?

Reply:
I have the same feeling... do we really need to sync both the property and the DOM text content? 

My guess is that the user would generally use just one of the ways: either set the helper text with the property or with the DOM content. In this regard, there is probably no necessity to fill the `helperText` property with the DOM text content.

Reply:
I think we can remove it after all. The initial idea was to keep those in sync but now it's rather confusing.

Reply:
Removed this logic, see 6af4d95